pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -134,7 +136,25 @@ func (r *ReconcileComplianceRemediation) reconcileRemediation(instance *compv1al
 	if utils.IsMachineConfig(instance.Spec.Object) {
 		return r.reconcileMcRemediation(instance, logger)
 	}
-	// TODO(jaosorior): Create general flow of applying remediations
+	return r.reconcileGenericRemediation(instance, logger)
+}
+
+// Gets a generic remediation and ensures the object exists in the cluster if the
+// remediation if applicable
+func (r *ReconcileComplianceRemediation) reconcileGenericRemediation(instance *compv1alpha1.ComplianceRemediation, logger logr.Logger) error {
+	if instance.Spec.Apply {
+		obj := instance.Spec.Object
+		found := obj.DeepCopy()
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
```

Comment:
Why make a deep copy of the object before getting?
Also, will this client work the same for a non-namespaced object?

Reply:
> Why make a deep copy of the object before getting?

Because it copies the GVK which is needed in order to get the object. Can't use unstructured or runtime.Object for this one straight.

> Also, will this client work the same for a non-namespaced object?

Yep